vernac/lemmas.ml

```diff
@@ -335,12 +335,28 @@ let initialize_named_context_for_proof () =
       let d = if variable_opacity id then NamedDecl.LocalAssum (id, NamedDecl.get_type d) else d in
       Environ.push_named_context_val d signv) sign Environ.empty_named_context_val
 
+(* Default proof mode, to be set at the beginning of proofs for
+   programs that cannot be statically classified. *)
+let default_proof_mode = ref "No"
+let get_default_proof_mode () = !default_proof_mode
+let set_default_proof_mode mn = default_proof_mode := mn
+
+let proof_mode_opt_name = ["Default";"Proof";"Mode"]
+let () =
+  Goptions.declare_string_option Goptions.{
+    optdepr = false;
+    optname = "default proof mode" ;
+    optkey = proof_mode_opt_name;
+    optread = get_default_proof_mode;
+    optwrite = set_default_proof_mode;
+  }
+
```

Comment:
It looks like this should live in `vernacentries`.

Reply:
Done.